src/features/organizations/components/OrganizationSwitcher/useOrgSwitcher.ts

```diff
@@ -0,0 +1,99 @@
+import Fuse from 'fuse.js';
+import { RootState } from 'core/store';
+import { TreeItemData } from 'features/organizations/types';
+import useLocalStorage from 'zui/hooks/useLocalStorage';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+function useOrgSwitcher(orgId: number, searchString: string) {
+  function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+    return value !== null && value !== undefined;
+  }
+
+  const treeDataList = useSelector(
+    (state: RootState) => state.organizations.treeDataList
+  );
+  const orgData = treeDataList.items.map((item) => item.data).filter(notEmpty);
+
+  const flatOrgData = makeFlatOrgData(orgData);
+
+  const [recentOrganizationIds, setRecentOrganizationIds] = useLocalStorage(
+    'recentOrganizationIds',
+    [] as number[]
+  );
+
+  function makeFlatOrgData(orgData: TreeItemData[]): TreeItemData[] {
+    let children = [] as TreeItemData[];
+    const flatOrgData = orgData.map((org) => {
+      if (org.children && org.children.length) {
+        children = [...children, ...org.children];
+      }
+      return org;
+    });
+
+    return flatOrgData.concat(
+      children.length ? makeFlatOrgData(children) : children
+    );
+  }
```

Comment:
Correct me if I'm wrong, but I think this could be moved outside of the `useOrgSwitcher`, just to make the hook smaller and easier to read.

Reply:
I'll make it into a separate utility function!

Reply:
Just to clarify, I don't think it should be considered reusable. It's still very tied to what's happening here. I just think it should be defined outside of `useOrgSwitcher` (the function) but still inside `useOrgSwitcher.ts` (the module).

Reply:
aa now i understand! thank you for clarifying! :D